From f7c60f74c063435a2fb2e5993b4cf2c32cf88fc5 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 25 Feb 2024 12:43:25 +1000
Subject: [PATCH] Remove deprecated distutils usage

---
 test/test_datasets_download.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/test_datasets_download.py b/test/test_datasets_download.py
index cf0211dbb1e..71aeb9ea774 100644
--- a/test/test_datasets_download.py
+++ b/test/test_datasets_download.py
@@ -1,12 +1,12 @@
 import contextlib
 import itertools
+import shutil
 import tempfile
 import time
 import traceback
 import unittest.mock
 import warnings
 from datetime import datetime
-from distutils import dir_util
 from os import path
 from urllib.error import HTTPError, URLError
 from urllib.parse import urlparse
@@ -180,7 +180,7 @@ def collect_urls(dataset_cls, *args, **kwargs):
 @pytest.fixture(scope="module", autouse=True)
 def root():
     yield ROOT
-    dir_util.remove_tree(ROOT)
+    shutil.rmtree(ROOT)
 
 
 def places365():
