From 21dc97fda8b6995a408f0e2919af78a70c8a7d3d Mon Sep 17 00:00:00 2001
From: Nicolas Hug <contact@nicolas-hug.com>
Date: Mon, 21 Mar 2022 11:43:57 +0000
Subject: [PATCH] More robust check in tests for 16 bits images

---
 test/test_image.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/test_image.py b/test/test_image.py
index f9e88e8ad..048f7c364 100644
--- a/test/test_image.py
+++ b/test/test_image.py
@@ -158,7 +158,7 @@ def test_decode_png(img_path, pil_mode, mode):
 
     img_pil = normalize_dimensions(img_pil)
 
-    if "16" in img_path:
+    if img_path.endswith("16.png"):
         # 16 bits image decoding is supported, but only as a private API
         # FIXME: see https://github.com/pytorch/vision/issues/4731 for potential solutions to making it public
         with pytest.raises(RuntimeError, match="At most 8-bit PNG images are supported"):
-- 
2.35.1.windows.2

