From e5aaad7f5584dd73a231ce94ec8a87160a3b4e48 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 3 Nov 2024 22:04:00 -0500
Subject: [PATCH 2/3] Force nvjpeg and force failure

---
 setup.py | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/setup.py b/setup.py
index 3b4e17a..e993e0c 100644
--- a/setup.py
+++ b/setup.py
@@ -335,17 +335,14 @@ def make_image_extension():
             warnings.warn("Building torchvision without WEBP support")
 
     if USE_NVJPEG and (torch.cuda.is_available() or FORCE_CUDA):
-        nvjpeg_found = CUDA_HOME is not None and (Path(CUDA_HOME) / "include/nvjpeg.h").exists()
-
-        if nvjpeg_found:
-            print("Building torchvision with NVJPEG image support")
-            libraries.append("nvjpeg")
-            define_macros += [("NVJPEG_FOUND", 1)]
-            Extension = CUDAExtension
-        else:
-            warnings.warn("Building torchvision without NVJPEG support")
+        # conda-forge supports cross compilation of nvjpeg whic isn't
+        # Really supported upstream
+        print("Building torchvision with NVJPEG image support")
+        libraries.append("nvjpeg")
+        define_macros += [("NVJPEG_FOUND", 1)]
+        Extension = CUDAExtension
     elif USE_NVJPEG:
-        warnings.warn("Building torchvision without NVJPEG support")
+        raise RuntimeError("NVJPEG Requested but building without it")
 
     return Extension(
         name="torchvision.image",
