diff --git a/test/test_transforms_v2.py b/test/test_transforms_v2.py
index f9218c3e84..012e556cae 100644
--- a/test/test_transforms_v2.py
+++ b/test/test_transforms_v2.py
@@ -53,15 +53,6 @@ from torchvision.transforms.v2.functional._geometry import _get_perspective_coef
 from torchvision.transforms.v2.functional._utils import _get_kernel, _register_kernel_internal


-# turns all warnings into errors for this module
-pytestmark = [pytest.mark.filterwarnings("error")]
-
-if sys.version_info[:2] >= (3, 12):
-    # torchscript relies on some AST stuff that got deprecated in 3.12,
-    # so we have to explicitly ignore those otherwise we'd error on warnings due to the pytestmark filter above.
-    pytestmark.append(pytest.mark.filterwarnings("ignore::DeprecationWarning"))
-
-
 @pytest.fixture(autouse=True)
 def fix_rng_seed():
     set_rng_seed(0)
